src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,556 @@
+
+               Input:
+                cb - callback variable of type ClimateMachine.GenericCallbacks.Every*
+               Updates:
+                Nothing
+               Returns:
+                Nothing - prints to REPL
+"""
+function scprintref(cb)
+    if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+        # Get print format lengths for cols 1 and 2 so they are aligned
+        # for readability.
+        phi = cb.func.curStats_flat
+        f = 1
+        a1l = maximum(length.(map(
+            i -> (phi[i])[f],
+            range(1, length = length(phi)),
+        )))
+        f = 2
+        a2l = maximum(length.(String.((map(
+            i -> (phi[i])[f],
+            range(1, length = length(phi)),
+        )))))
+        fmt1 = @sprintf("%%%d.%ds", a1l, a1l) # Column 1
+        fmt2 = @sprintf("%%%d.%ds", a2l, a2l) # Column 2
+        fmt3 = @sprintf("%%28.20e")         # All numbers at full precision
+        # Create an string of spaces to be used for fomatting
+        sp = "                                                                           "
+
+        # Write header
+        println("# BEGIN SCPRINT")
+        println("# varr - reference values (from reference run)    ")
+        println("# parr - digits match precision (hand edit as needed) ")
+        println("#")
+        println("# [")
+        println("#  [ MPIStateArray Name, Field Name, Maximum, Minimum, Mean, Standard Deviation ],")
+        println("#  [         :                :          :        :      :          :           ],")
+        println("# ]")
+        #
+        # Write tables
+        #  Reference value and precision match tables are separate since it is more
+        #  common to update reference values occiasionally while precision values are
+        #  typically changed rarely and the precision values are hand edited from experience.
+        #
+        # Write table of reference values
+        println("varr = [")
+        for lv in cb.func.curStats_flat
+            s1 = lv[1]
+            l1 = length(s1)
+            s1 = sp[1:(a1l - l1)] * "\"" * s1 * "\""
+            s2 = lv[2]
+            if typeof(s2) == String
+                l2 = length(s2)
+                s2 = sp[1:(a2l - l2)] * "\"" * s2 * "\""
+                s22 = ""
+            end
+            if typeof(s2) == Symbol
+                s22 = s2
+                l2 = length(String(s2))
+                s2 = sp[1:(a2l - l2 + 1)] * ":"
+            end
+            s3 = sprintf1(fmt3, lv[3])
+            s4 = sprintf1(fmt3, lv[4])
+            s5 = sprintf1(fmt3, lv[5])
+            s6 = sprintf1(fmt3, lv[6])
+            println(
+                " [ ",
+                s1,
+                ", ",
+                s2,
+                s22,
+                ",",
+                s3,
+                ",",
+                s4,
+                ",",
+                s5,
+                ",",
+                s6,
+                " ],",
+            )
+        end
```

Comment:
Can/should we use something like [PrettyTables.jl](https://github.com/ronisbr/PrettyTables.jl) here and below? @simonbyrne @kpamnany